gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -61,17 +64,19 @@ def initialize(method:, method_name:, raw_arg_types:, raw_return_type:, bind:, m
     # If sig params are declared but there is a single parameter with a missing name
     # **and** the method ends with a "=", assume it is a writer method generated
     # by attr_writer or attr_accessor
-    writer_method = declared_param_names != [nil] && parameters == [[:req]] && method_name[-1] == "="
+    writer_method = declared_param_names != SIG_EMPTY_DECLARED_PARAMETERS && parameters == UNNAMED_REQUIRED_PARAMETERS && method_name[-1] == "="
     # For writer methods, map the single parameter to the method name without the "=" at the end
     parameters = [[:req, method_name[0...-1].to_sym]] if writer_method
     param_names = parameters.map {|_, name| name}
     missing_names = param_names - declared_param_names
```

Comment:
might be minor: do we need to materialize this here? Seems like we only care about whether it's non-empty initially and only need to materialize if so

Could maybe save array allocation if not having missing params is the common case?

Reply:
This is an interesting idea!  Maybe we could avoid materializing `param_names` altogether if we were willing to write out some loops manually.